aiosmtpd/smtp.py

```diff
@@ -449,17 +657,21 @@ def _getparams(self, params):
         return result
 
     def _syntax_available(self, method):
-        if getattr(method, '__smtp_syntax__', None) is None:
+        if not hasattr(method, '__smtp_syntax__'):
```

Comment:
Since we're not using the result of this particular `getattr()` anyways, why not just use `hasattr()`?

Reply:
Can we check the timing of this, over:

```
try:
    return bool(getattr(self, method.__smtp_syntax_when__))
except AttributeError:
    return False
else:
    return True
```

Reply:
If there's a significant difference we should do the faster one, but otherwise, I'm cool with this current change.

Reply:
Gut feeling says it's about the same 😁. But let me do some `timeit`s and I'll get back to you.

Reply:
[Checkout this gist](https://gist.github.com/pepoluan/908996ede1da06c0058061db40d89016)

Surprisingly, the one with `hasattr` is nearly **twice as fast** as the one with `try ... except` 😲

Based on this result, I will use the current implementation with `hasattr`